pkg/util/proxy/proxy.go

```diff
@@ -124,7 +124,7 @@ func NewThrottledUpgradeAwareProxyHandler(location *url.URL, transport http.Roun
 
 func GetTlsConfigForCluster(ctx context.Context, cluster *clusterapis.Cluster, secretGetter SecretGetterFunc) (*tls.Config, error) {
 	// The secret is optional for a pull-mode cluster, if no secret just returns a config with root CA unset.
-	if cluster.Spec.SecretRef == nil {
+	if cluster.Spec.SecretRef == nil || cluster.Spec.InsecureSkipTLSVerification {
```

Comment:
Should we set the RootCAs even if InsecureSkipTLSVerification is true(If the caBundle exists)?

Reply:
> Should we set the RootCAs even if InsecureSkipTLSVerification is true(If the caBundle exists)?

en.., sure, It's a batter choice

Reply:
/cc @jwcesign